lib/active_record_shards/association_scope-4-0.rb

```diff
@@ -0,0 +1,19 @@
+module ActiveRecordShards
+  module AssociationScope
+    def self.included(base)
+      base.send(:alias_method, :bind_without_default_shard, :bind)
+      base.send(:alias_method, :bind, :bind_with_default_shard)
+    end
+
+    def bind_with_default_shard(scope, table_name, column_name, value, tracker)
```

Comment:
prefer `*args` to not break this in weird ways

Reply:
I actually need `scope` and `table_name`. Also this is only for Rails 4.2 which won't be changing anymore so risk is pretty low.